client/js/heat.js

```diff
@@ -0,0 +1,91 @@
+class HeatMapRow {
+  /**
+   * format for a row for heat map values obj
+   * @param {number[]} data The array of 0s and 1s for a row in a heat map
+   */
+  constructor(data) {
+    this.data = data;
+  }
+}
+
+// data will be from backend next time
+const TOTAL_LIKED = 13;
+// element value of x means xth latest video is music
+const LIKED_MUSIC_HIST = [1, 4, 7];
+
+// ith value of 1 means ith latest video is music
```

Comment:
This comment is difficult to parse. It sounds like this is an array of binary values (versus a binary tree being stored in an array, which is what I expect a "binary array" to be). If this is the case, store booleans instead where you can be more clear:
```
const likedMusicHist = markMusicDays(LIKED_MUSIC_HIST, TOTAL_LIKED);
```

Reply:
I don't want to store booleans because the chart can only render integer values (what I have) not coolean values

Reply:
Then leave a comment detailing this. Otherwise it's odd that you're representing a boolean values as a list of numbers. Something like:
```
// The histogram library takes a list of numbers so that heavier music
// days are displayed as darker. We don't want this gradient, so a day with
// any amount of music will be stored as a `1`.
```